src/Compiler/CompilerOptions.cpp

```diff
@@ -1073,6 +1087,17 @@ void initCompilerConfig() {
             : std::vector<std::string>{"jniruntime", "cruntime"});
     addCompilerConfig(CCM_SHARED_LIB_PATH_DEPS, {getLibraryPath()});
 
+    if (enableParallel) {
+      // Add the static and dynamic path to OMP lib, which resides with the llvm
+      // tools. Thus fetch library path associated with "llc".
+      std::string libPath = getToolLibraryPath("llc");
+      addCompilerConfig(CCM_SHARED_LIB_DEPS, std::vector<std::string>{"omp"});
+      addCompilerConfig(
+          CCM_SHARED_LIB_PATH_DEPS, std::vector<std::string>{libPath});
+      addCompilerConfig(
+          CCM_SHARED_DYN_LIB_PATH_DEPS, std::vector<std::string>{libPath});
+    }
```

Comment:
For the libPath, we add a `-L` for each element of the list. For the dynamic library path, we need this prefix: `-Wl,-rpath,`

Reply:
No. -L works for both static and shared libs. `-Wl,-rpath` just embeds the path into the executable. It's generally a bad idea because the path is tied to the specific build environment and only valid at compile time. As soon as the lib get installed or moved to a different place, it stops working. To find the shared lib at runtime, you typically use `LD_LIBRARY_PATH` if the lib is not installed in the "standard" place.

Reply:
Thanks, that is very relevant point; will experiment with removing the `-Wl,-rpath` and use the `LD_LIBRARY_PATH` instead.